src/Utils/Utils.php

```diff
@@ -169,11 +165,19 @@ public static function shouldHandleRequest(
         // Allow request if shop is not staging or is staging with findologic=on flag set
         $allowRequestForStaging = (!$isStagingShop || ($isStagingShop && $isStagingSession));
 
+        if ($config->isActive() && ($isDirectIntegration || $allowRequestForStaging)) {
+            $findologicService->enableSmartSuggest();
+        }
+
+        if (!$config->isActive() || ($isCategoryPage && !$config->isActiveOnCategoryPages())) {
+            return $findologicService->disable();
+        }
+
         if ($isDirectIntegration || !$allowRequestForStaging) {
-            return $findologicService->setDisabled();
+            return $findologicService->disable();
```

Comment:
No the Smart Suggest has an own path. It is disabled by default, we only activate it in a specific case. That would also be the case for the Findologic service, since that one is also disabled by default and only enabled if really needed. That being said, we could also just return the current state here instead of explicitly manipulating the state, as that wouldn't change the outcome.

Reply:
Alright